### PR TITLE
feat/playground-second-iteration

### PR DESCRIPTION
**Pull request by FredrikOseberg**
This contains fixes and improvements to the new playground feature:
* The playground will now search for feature toggles automatically if the URL contains a valid query
* The playground defaults to the first available valid environment
* Moved context form above the JSON field
* Fixed a bug where placeholder would display in search
* Changed the layout to keep the results on the side
* Added date input for currentTime and legal values dropdown
* Added sticky form
* Added variant popover information
* Added ability to search for isEnabled result
* Added CodeMirror JSON editor

### CODE REVIEW

File: src/component/playground/Playground/Playground.tsx [hunk 1]
```diff
@@ -103,88 +106,115 @@ export const Playground: VFC<IPlaygroundProps> = () => {
...
+        }
+
+        if (results && !matches) {
+            return '65%';
+        }
+
+        return '50%';
+    };
+
+    const resolveLayout = () => {
```
Comment (Tymek) [inline src/component/playground/Playground/Playground.tsx:155]: I think this can be inlined. `sx={{ display: 'flex', flexDirection: !matches ? 'row' : 'column' }}`

File: src/component/playground/Playground/PlaygroundForm/PlaygroundCodeFieldset/PlaygroundEditor/PlaygroundEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+interface IPlaygroundEditorProps {
+    context: string | undefined;
+    setContext: Dispatch<SetStateAction<string | undefined>>;
+    error: string | undefined;
+}
+
+const StyledEditorHeader = styled('aside')(({ theme }) => ({
+    height: '50px',
+    backgroundColor: theme.palette.grey[100],
+    borderTopRightRadius: '5px',
```
Comment (Tymek) [inline src/component/playground/Playground/PlaygroundForm/PlaygroundCodeFieldset/PlaygroundEditor/PlaygroundEditor.tsx:18]: border radius has values in `theme`
(`theme.shape.`, borderRadius, borderRadiusMedium, borderRadiusLarge, borderRadiusExtraLarge)

Comment (FredrikOseberg) [reply src/component/playground/Playground/PlaygroundForm/PlaygroundCodeFieldset/PlaygroundEditor/PlaygroundEditor.tsx:18]: Thanks!

File: src/component/playground/Playground/Playground.tsx (hunk 1 above)
Comment (FredrikOseberg) [reply src/component/playground/Playground/Playground.tsx:155]: Agree

File: src/component/common/Search/Search.tsx [hunk 3]
```diff
@@ -59,7 +61,7 @@ export const Search = ({
     const placeholder = `${customPlaceholder ?? 'Search'} (${hotkey})`;
 
     return (
-        <div className={styles.container}>
+        <div className={styles.container} style={containerStyles}>
```
[the remark this answers is not in the document]
Comment (FredrikOseberg) [reply src/component/common/Search/Search.tsx:64]: I do agree, and I'm not entirely happy with how it is now in terms of Styled components / sx / style mix in the project now. But I find using classes is difficult now because (a) we'd like to move away from makeStyles since that is now a third party package that may or may not be maintained in the future, and (b) we need our styles to be connected to the theme, which we can't access through regular styles sheets or css modules.

File: src/component/playground/Playground/Playground.tsx [hunk 4]
```diff
@@ -103,88 +106,114 @@ export const Playground: VFC<IPlaygroundProps> = () => {
...
+        }
+
+        if (results && !matches) {
+            return '65%';
+        }
+
+        return '50%';
+    };
+
+    const formWidth = resolveFormWidth();
```
Comment (Tymek) [inline src/component/playground/Playground/Playground.tsx:155]:
```suggestion
    const formWidth = results && !matches ? '35%' : 'auto';
```

Comment (Tymek) [reply src/component/playground/Playground/Playground.tsx:155]: Same with `resultsWidth`. I think this can be less verbose.

**Review by Tymek — COMMENTED** [review]
I don't think helper functions are needed for `formWidth`. Also, `borderRadius` from theme wasn't updated.

File: src/component/playground/Playground/Playground.tsx (hunk 4 above)
Comment (FredrikOseberg) [reply src/component/playground/Playground/Playground.tsx]: I agree. Refactored formWidth to use ternary. I tried a double ternary for resultsWidth but I found it really hard to read so instead I parameterized the helper and moved it out of the way of the component.

**Review by Tymek — APPROVED** [review]
LGTM

File: src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { Typography, Box, Divider } from '@mui/material';
+import { PlaygroundGuidanceSection } from './PlaygroundGuidanceSection/PlaygroundGuidanceSection';
+
+export const PlaygroundGuidance = () => {
+    return (
+        <Box sx={{ ml: 4 }}>
+            <Typography variant="body1">
+                Unleash playground is for helping you to undestand how unleash
+                works, how feature toggles are evaluated and for you to easily
+                debug your feature toggles.
```
Comment (thomasheartman) [inline src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx:10]: How about something like this? You might need to adjust the indentation

```suggestion
		The Unleash playground was designed to help you understand how Unleash
		works. You can use it to debug your current feature toggle setup or to
		experiment with how application context impacts feature toggle
		evaluation.
```

File: src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+            <Divider sx={{ mt: 2, mb: 2 }} />
+
+            <Typography variant="body1" sx={{ mb: 1 }}>
+                What you need to do is:
+            </Typography>
+
+            <PlaygroundGuidanceSection
+                headerText="  Select in which environment you want to test your
+                            feature toggle configuration"
```
Comment (thomasheartman) [inline src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx:21]:
```suggestion
                headerText="Select an environment to test your
                            feature toggle configuration in"
```

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Nice work! I suggest a few minor text changes

File: src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                bodyText="  You can also specify specific projects, or check
+                            toggles in all projects."
+                sectionNumber="1"
+            />
+
+            <PlaygroundGuidanceSection
+                headerText="Select a context field that you'd like to check"
+                bodyText="Configure multiple context fields or leave empty to test against an empty context"
```
Comment (thomasheartman) [inline src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx:29]: When you test against an empty context, we automatically apply `appName`, though. Should that be mentioned?

```suggestion
                bodyText="You can configure as many context fields context fields as you want. You can also leave the context empty to test against an empty context."
```

File: src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                sectionNumber="2"
+            />
+
+            <PlaygroundGuidanceSection
+                headerText="Try configuration"
```
Comment (thomasheartman) [inline src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx:34]:
```suggestion
                headerText="Submit the form to try the configuration"
                bodyText="The results of evaluating your feature toggles will appear after you submit the form."
```

File: src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                sectionNumber="3"
+            />
+
+            <PlaygroundGuidanceSection
+                headerText="View the results"
+                sectionNumber="4"
+            />
```
Comment (thomasheartman) [inline src/component/playground/Playground/PlaygroundGuidance/PlaygroundGuidance.tsx:41]: Suggest we move this into the previous step. It's not really a step per se.

**Review by thomasheartman — CHANGES_REQUESTED** [review]
A few more text suggestions for the guidance. Think this might be clearer 💁🏼
